apps/prairielearn/src/lib/db-types.ts

```diff
@@ -585,6 +585,16 @@ export const GradingJobSchema = z.object({
 });
 export type GradingJob = z.infer<typeof GradingJobSchema>;
 
+export const ClientFingerprintSchema = z.object({
+  id: IdSchema,
+  user_id: IdSchema,
+  user_session_id: IdSchema.nullable(),
```

Comment:
I agree that it would be great to fix this. I think that using sessions for authn is the way to go here -- that makes it clear that if you are authenticated then you have a session, and the auth pages themselves are always going to be special in any case so they can just deal with the session weirdness.

Needless to say, we shouldn't worry about this for now, however.

Reply:
The only reason I'm proposing worrying about this now is the sweet sweet `NOT NULL` constraint we could add to `user_session_id`. If you're fine to forgo that, I'm happy to wait for now.

Reply:
Ooooo, you know I'm a sucker for a NOT NULL. I'm not 100% on board with changing sessions to write them immediately to the DB so we can get the ID. @nwalters512 can you coordinate this with Myles?

Reply:
Sorry, to check: you're **not** 100% on board, or is that a typo?

Reply:
Ha, sorry again, I fixed that comment just after you replied. I *am* 100% on board 😄